src/transformers/generation_tf_utils.py

```diff
@@ -650,31 +650,54 @@ def generate(
-        assert isinstance(early_stopping, bool), "`early_stopping` should be a boolean."
-        assert isinstance(num_beams, int) and num_beams > 0, "`num_beams` should be a strictly positive integer."
-        assert temperature > 0, "`temperature` should be strictly positive."
-        assert isinstance(top_k, int) and top_k >= 0, "`top_k` should be a positive integer."
-        assert 0 <= top_p <= 1, "`top_p` should be between 0 and 1."
-        assert repetition_penalty >= 1.0, "`repetition_penalty` should be >= 1."
-        assert input_ids is not None or (
-            isinstance(bos_token_id, int) and bos_token_id >= 0
-        ), "If input_ids is not defined, `bos_token_id` should be a positive integer."
-        assert pad_token_id is None or (
-            isinstance(pad_token_id, int) and (pad_token_id >= 0)
-        ), "`pad_token_id` should be a positive integer."
-        assert (eos_token_id is None) or (
-            isinstance(eos_token_id, int) and (eos_token_id >= 0)
-        ), "`eos_token_id` should be a positive integer."
-        assert length_penalty > 0, "`length_penalty` should be strictly positive."
-        assert (
-            isinstance(num_return_sequences, int) and num_return_sequences > 0
-        ), "`num_return_sequences` should be a strictly positive integer."
-        assert (
-            bad_words_ids is None or isinstance(bad_words_ids, list) and isinstance(bad_words_ids[0], list)
-        ), "`bad_words_ids` is either `None` or a list of lists of tokens that should not be generated"
+        if not isinstance(max_length, int) or max_length <= 0:
+            raise ValueError("`max_length` should be a strictly positive integer.")
+
+        if not isinstance(min_length, int) or min_length < 0:
+            raise ValueError("`min_length` should be a positive integer.")
+
+        if not isinstance(do_sample, bool):
+            raise TypeError("`do_sample` should be a boolean.")
+
+        if not isinstance(early_stopping, bool):
+            raise TypeError("`early_stopping` should be a boolean.")
+
+        if not isinstance(num_beams, int) or num_beams <= 0:
+            raise ValueError("`num_beams` should be a strictly positive integer.")
+
+        if temperature <= 0:
+            raise ValueError("`temperature` should be strictly positive.")
+
+        if not isinstance(top_k, int) or top_k < 0:
+            raise ValueError("`top_k` should be a positive integer.")
+
+        if 0 > top_p or top_p > 1:
+            raise ValueError("`top_p` should be between 0 and 1.")
+
+        if repetition_penalty < 1.0:
+            raise ValueError("`repetition_penalty` should be >= 1.")
+
+        if input_ids is None and (not isinstance(bos_token_id, int) or bos_token_id < 0):
+            raise ValueError("If input_ids is not defined, `bos_token_id` should be a positive integer.")
+
+        if pad_token_id is not None and (not isinstance(pad_token_id, int) or not pad_token_id < 0):
```

Comment:
```suggestion
        if pad_token_id is not None and (not isinstance(pad_token_id, int) or pad_token_id < 0):
```
Too many negations here ;-)